diskquota.c

```
@@ -1029,3 +1015,45 @@ is_valid_dbid(Oid dbid)
 	ReleaseSysCache(tuple);
 	return true;
 }
+
+bool
+worker_increase_epoch(Oid database_oid)
+{
+	LWLockAcquire(diskquota_locks.worker_map_lock, LW_EXCLUSIVE);
+
+	bool found = false;
+	DiskQuotaWorkerEntry * workerentry = (DiskQuotaWorkerEntry *) hash_search(
+		disk_quota_worker_map, (void *) &database_oid, HASH_FIND, &found);
+	
+	if (found)
+	{
+		++(workerentry->epoch);
+	}
+	LWLockRelease(diskquota_locks.worker_map_lock);
+	return found;
+}
+
+unsigned int
+worker_get_epoch(Oid database_oid)
+{
+	LWLockAcquire(diskquota_locks.worker_map_lock, LW_SHARED);
+
+	bool found = false;
+	unsigned int epoch = 0;
+	DiskQuotaWorkerEntry * workerentry = (DiskQuotaWorkerEntry *) hash_search(
+		disk_quota_worker_map, (void *) &database_oid, HASH_FIND, &found);
+	
+	if (found)
+	{
+		epoch = workerentry->epoch;
+	}
+	LWLockRelease(diskquota_locks.worker_map_lock);
```

Otherwise should throw an error.